circuits/src/programs/mod.rs

```diff
@@ -101,7 +97,11 @@ impl<E: Environment> Memory<E> {
         match self.registers.get(locator.0 as usize) {
             Some(register) => match register.get().is_some() {
                 true => panic!("Register {} is already set", locator.0),
-                false => register.set(value.clone()),
+                false => {
```

Comment:
The only non-trivial change seems to be this one; was this intended? If so, the rest LGTM